### PR TITLE
backend/A64: Fix SpinLockImpl not setting required memory permissions.

### PR DESCRIPTION
**Pull request by Steveice10**
Fixes the SpinLockImpl code emitting not setting RW before and RX after writing. Resolves issue where Citra would crash on start when built for Apple Silicon.

As part of this change I moved the memory protect/unprotect code from BlockOfCode to the base CodeBlock so it would be available in ARM64CodeBlock.

With this fix and the Citra Vulkan PR I was able to boot and play Pokemon Moon on a native ARM64 build of Citra.

### CODE REVIEW

**Comment by Steveice10** [comment]
Fixed introduced clang-format errors

File: src/dynarmic/backend/A64/block_of_code.cpp [hunk 1]
```diff
@@ -53,23 +53,6 @@ namespace {
 constexpr size_t TOTAL_CODE_SIZE = 128 * 1024 * 1024;
 constexpr size_t FAR_CODE_OFFSET = 100 * 1024 * 1024;
 
-#ifdef DYNARMIC_ENABLE_NO_EXECUTE_SUPPORT
```
Comment (SachinVin) [inline src/dynarmic/backend/A64/block_of_code.cpp:56]: Could you also move the headers form here to code_block.h

Comment (Steveice10) [reply src/dynarmic/backend/A64/block_of_code.cpp:56]: Assuming you mean the platform header includes, done.

Comment (SachinVin) [reply src/dynarmic/backend/A64/block_of_code.cpp:56]: Yep, thanks!

**Pull request merged**